Program crashes when using the package on windows x86

at the moment the app crashes here
const System = importClass('java.lang.System'); 

npm ERR! code ELIFECYCLE 
npm ERR! errno 3221225477 
....

java should be available
PS D:\workspace\electron_example\electron-quick-start> java -version openjdk version "11.0.17" 2022-10-18 OpenJDK Runtime Environment Temurin-11.0.17+8 ...

some additional version infos
Oracle Java 32bit
NodeJS 16.15.0  32bit
Electron 21.2.3
Windows 10

any ideas?

greetings Klaus

_Originally posted by @tripower in https://github.com/MarkusJx/node-java-bridge/discussions/32#discussioncomment-4165137_

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -80,6 +80,7 @@ jobs:
               timeout-minutes: 20
             - name: Create package
               if: ${{matrix.node-version == '18.x'}}
+              shell: bash
               run: |
                   node -e 'let p=require("./package.json");p.files.push("dist/*.node");require("fs").writeFileSync("package.json",JSON.stringify(p,null,4))'
                   npm pack

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -24,8 +24,31 @@ jobs:
     test-npm:
         strategy:
             matrix:
-                node-version: [16.x, 18.x]
-                os: [windows-latest, ubuntu-latest, macOs-latest]
+                include:
+                    - os: ubuntu-latest
+                      node-version: 16.x
+                      arch: x64
+                    - os: ubuntu-latest
+                      node-version: 18.x
+                      arch: x64
+                    - os: windows-latest
+                      node-version: 16.x
+                      arch: x64
+                    - os: windows-latest
+                      node-version: 18.x
+                      arch: x64
+                    - os: windows-latest
+                      node-version: 16.x
+                      arch: x86
+                    - os: windows-latest
+                      node-version: 18.x
+                      arch: x86
+                    - os: macos-latest
+                      node-version: 16.x
+                      arch: x64
+                    - os: macos-latest
+                      node-version: 18.x
+                      arch: x64
         runs-on: ${{matrix.os}}
         steps:
             - uses: actions/checkout@v3
@@ -38,10 +61,16 @@ jobs:
               with:
                   distribution: 'temurin'
                   java-version: '17'
+                  architecture: ${{matrix.arch}}
             - name: Rust Cache
               uses: Swatinem/rust-cache@v2.0.0
             - name: Install
               run: npm ci
+            - name: Build (win32)
+              if: matrix.os == 'windows-latest' && matrix.arch == 'x86'
+              run: |
+                  rustup target add i686-pc-windows-msvc
+                  npm run build -- -- -- --target i686-pc-windows-msvc
             - name: Run tests
               run: npm test
               timeout-minutes: 20
@@ -54,7 +83,7 @@ jobs:
               uses: actions/upload-artifact@v3
               if: ${{matrix.node-version == '18.x'}}
               with:
-                  name: java-x64-${{matrix.os}}
+                  name: java-${{matrix.arch}}-${{matrix.os}}
                   path: java-bridge-*.tgz
 
     test-linux-aarch64-gnu-binding:

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -67,13 +67,16 @@ jobs:
               uses: Swatinem/rust-cache@v2.0.0
             - name: Install
               run: npm ci
+            - name: Build
+              if: ${{matrix.os != 'windows-latest' || matrix.arch == 'x64'}}
+              run: npm run build
             - name: Build (win32)
-              if: matrix.os == 'windows-latest' && matrix.arch == 'x86'
+              if: ${{matrix.os == 'windows-latest' && matrix.arch == 'x86'}}
               run: |
                   rustup target add i686-pc-windows-msvc
                   npm run build -- -- --target i686-pc-windows-msvc
             - name: Run tests
-              run: npm test
+              run: npm run testOnly
               timeout-minutes: 20
             - name: Create package
               if: ${{matrix.node-version == '18.x'}}

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -71,7 +71,7 @@ jobs:
               if: matrix.os == 'windows-latest' && matrix.arch == 'x86'
               run: |
                   rustup target add i686-pc-windows-msvc
-                  npm run build -- -- -- --target i686-pc-windows-msvc
+                  npm run build -- -- --target i686-pc-windows-msvc
             - name: Run tests
               run: npm test
               timeout-minutes: 20

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -56,6 +56,7 @@ jobs:
               uses: actions/setup-node@v3.4.1
               with:
                   node-version: ${{matrix.node-version}}
+                  architecture: ${{matrix.arch}}
             - name: Setup java 17
               uses: actions/setup-java@v3
               with:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,7 +1,7 @@
 [package]
 edition = "2021"
 name = "java"
-version = "2.1.6"
+version = "2.1.7"
 
 [lib]
 crate-type = ["cdylib"]

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,6 +20,3 @@ rand = "0.8.5"
 napi-build = "2.0.1"
 bindgen = "0.60.1"
 java-locator = "0.1.2"
-
-[profile.release]
-lto = true

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
     "name": "java-bridge",
-    "version": "2.1.6",
+    "version": "2.1.7",
     "main": "dist/index.prod.min.js",
     "types": "dist/ts-src/index.d.ts",
     "description": "A bridge between Node.js and Java APIs",
@@ -70,12 +70,12 @@
         "chalk": "^5.1.2",
         "glob": "^8.0.3",
         "ora": "^6.1.2",
-        "typescript": "^4.8.4",
-        "yargs": "^17.6.1"
+        "typescript": "^4.9.3",
+        "yargs": "^17.6.2"
     },
     "devDependencies": {
-        "@napi-rs/cli": "^2.12.0",
-        "@types/chai": "^4.3.3",
+        "@napi-rs/cli": "^2.12.1",
+        "@types/chai": "^4.3.4",
         "@types/glob": "^8.0.0",
         "@types/is-ci": "^3.0.0",
         "@types/mocha": "^10.0.0",
@@ -83,7 +83,7 @@
         "@types/semver": "^7.3.13",
         "@types/webpack-node-externals": "^2.5.3",
         "@types/yargs": "^17.0.13",
-        "chai": "^4.3.6",
+        "chai": "^4.3.7",
         "cpy-cli": "^4.2.0",
         "expose-gc": "^1.0.0",
         "is-ci": "^3.0.1",
@@ -98,9 +98,9 @@
         "ts-loader": "^9.4.1",
         "ts-node": "^10.9.1",
         "tslib": "^2.4.1",
-        "typedoc": "^0.23.20",
-        "webpack": "^5.74.0",
-        "webpack-cli": "^4.10.0",
+        "typedoc": "^0.23.21",
+        "webpack": "^5.75.0",
+        "webpack-cli": "^5.0.0",
         "webpack-node-externals": "^3.0.0"
     },
     "optionalDependencies": {

--- a/test/testUtil.ts
+++ b/test/testUtil.ts
@@ -2,3 +2,6 @@ import isCi from 'is-ci';
 
 export const shouldIncreaseTimeout =
     isCi && (process.arch === 'arm64' || process.arch === 'arm');
+
+console.log('Process arch:', process.arch);
+console.log('Process platform:', process.platform);

